packages/react-pages/src/node/dynamic-modules/pages.ts

```diff
@@ -10,12 +10,9 @@ export async function renderPageList(pagesData: PagesData, isBuild: boolean) {
         // so we change the sub path
         subPath = '/__index'
       }
-      const dataProperty = isBuild
-        ? `data = () => import("/@react-pages/pages${subPath}")`
-        : `dataPath = "/@react-pages/pages${subPath}"`
       const code = `
 pages["${pageId}"] = {};
-pages["${pageId}"].${dataProperty};
+pages["${pageId}"].data = () => import("/@react-pages/pages${subPath}");
```

Comment:
We should write `import("/@react-pages/pages${subPath}")`
so that vite know this module is imported.
Otherwise hmr won't work. It always do full reload. related vite code: https://github.com/vitejs/vite/blob/63bd2502781b87a2c04a375d9e7b770f63d8857c/packages/vite/src/node/server/hmr.ts#L207

Reply:
That's weird, I haven't experienced this issue. You have?

We can export both `data` and `dataPath` in serve mode, so Jotai still works as expected.

Reply:
Yes. Previously vite can't analyse the import chain from this line: https://github.com/vitejs/vite-plugin-react-pages/blob/7eb042bae1c8eb6da59a466ce68b7d06ee56bf89/packages/react-pages/src/client/state.ts#L113
So when `/@react-pages/pages/page1` is updated, vite just think: "I don't know any module that imports `/@react-pages/pages/page1`, so I have to do full reload". Related vite code: https://github.com/vitejs/vite/blob/34ce760e51b0bb6d21e9496a9ea24571c499771c/packages/vite/src/node/server/hmr.ts#L135

Reply:
> We can export both data and dataPath in serve mode, so Jotai still works as expected.

Done. In [this commit](https://github.com/vitejs/vite-plugin-react-pages/pull/19/commits/e9911c246559178d43e1344dcf6941dd3ea0bd2c).